inbm/dispatcher-agent/dispatcher/xmlhandler.py

```diff
@@ -242,8 +242,8 @@ def get_root_elements(self, key: str, attr: str) -> list:
             raise XmlException(f"ERROR while fetching elements from root : {e}")
```

Comment:
**Suggestion:** The get_root_elements method should handle the case where the attribute does not exist on an element to prevent a TypeError. It should return None or a default value for such cases. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            for ele in self._root.findall(key):
                val = ele.get(attr, None)  # Provide a default value of None if the attribute is not found
                if val is not None:
                    elements.append(val)
            return elements
```